pkg/ingress/config/config.go

```diff
@@ -22,29 +22,37 @@ import (
 
 // Config struct contains ingress controller configuration
 type Config struct {
-	Kubernetes kubeConfig
-	OpenStack  osConfig
-	Octavia    octaviaConfig
+	Kubernetes kubeConfig    `mapstructure:"kubernetes"`
+	OpenStack  osConfig      `mapstructure:"openstack"`
+	Octavia    octaviaConfig `mapstructure:"octavia"`
 }
 
+// Configuration for connecting to Kubernetes API server, either api_host or kubeconfig should be configured.
 type kubeConfig struct {
-	ApiserverHost string
-	KubeConfig    string
+	// (Optional)Kubernetes API server host address.
+	ApiserverHost string `mapstructure:"api_host"`
+
+	// (Optional)Kubeconfig file used to connect to Kubernetes cluster.
+	KubeConfig string `mapstructure:"kubeconfig"`
 }
 
+// OpenStack credentials configuration, the section is required.
 type osConfig struct {
 	Username  string
 	Password  string
-	ProjectID string
-	AuthURL   string
+	ProjectID string `mapstructure:"project_id"`
+	AuthURL   string `mapstructure:"auth_url"`
 	Region    string
 }
 
+// Octavia service related configuration
 type octaviaConfig struct {
-	SubnetID           string
-	NodeSubnetID       string
-	AllocateFloatingIP bool
```

Comment:
Hello, what is the difference between SubnetID and NodeSubnetID ? Because it looks like your are removing completely NodeSubnetID. Thank you

Reply:
The `NodeSubnetID` is copied from the loadbalancer type service implementation of the cloud provider, it's not actually used in octavia ingress controller (at least for now), so removed.